### PR TITLE
feat(ui): Add frontend part for notification tab

### PR DESCRIPTION
**Pull request by Jordan91**
## Description

Video descibe behavior expected :
https://www.loom.com/share/dc8e1d237fb34ebea4ae05fe8b9022e0

Add new tab in the Resource Status Detail page of a resource (Host, Service, Metaservice).

If notification are disabled for a contact/contact group, this should be clearly notified. ( Same icon like in RS column Notification)

If no contacts/contact groups, a message “no contact/contact group is configured for this resource” is displayed.

Notification section shall contain 2 sub categories :

-Contacts
-Contact Groups 

In all cases, contact info to display are name and email address. Contact group info to display is contact group name
Both contact and contact group name are clickable and redirect to the configuration page of the related resource

ACL should be taken in account (if user is not entitled to see conf, the name should not be clickable and tooltip should be displayable “you are not entitled to access configuration”)

From there, browser back shall redirect to the previous RS page with resource details displayed.

## Type of change

- [x] New functionality (non-breaking change)

## Target serie

- [x] 22.04.x (master)

### CODE REVIEW

File: www/front_src/src/Resources/Details/index.test.tsx [hunk 1]
```diff
@@ -545,7 +546,7 @@ const mockedLocalStorageSetItem = jest.fn();
 Storage.prototype.getItem = mockedLocalStorageGetItem;
 Storage.prototype.setItem = mockedLocalStorageSetItem;
 
-describe(Details, () => {
+describe(Details, async () => {
```
Comment (Thebarda) [inline www/front_src/src/Resources/Details/index.test.tsx:549]:
```suggestion
describe(Details, () => {
```

File: www/front_src/src/Resources/Details/tabs/Notifications/Contacts.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        <span />
+
+        <Divider sx={{ gridColumn: '1 / -1' }} />
+      </>
+      {contacts.map((contact) => {
+        return (
+          <>
+            {getColumns(contact)}
+            {getConfigurationColumn(contact)}
+          </>
```
Comment (Thebarda) [inline www/front_src/src/Resources/Details/tabs/Notifications/Contacts.tsx:65]:
```suggestion
          <React.Fragment key={contact.alias}>
            {getColumns(contact)}
            {getConfigurationColumn(contact)}
          </React.Fragment>
```

File: www/front_src/src/Resources/Details/tabs/Notifications/Notification.tsx [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    request: getData,
+  });
+  const details = useAtomValue(detailsAtom);
+
+  const endpoint = path(['links', 'endpoints', 'notification_policy'], details);
+
+  const loadNotificationContacts = (): void => {
+    sendRequest({ endpoint }).then((retrievedNotificationContacts) => {
+      setNotificationContacts(retrievedNotificationContacts);
+    });
```
Comment (Thebarda) [inline www/front_src/src/Resources/Details/tabs/Notifications/Notification.tsx:45]:
```suggestion
    sendRequest({ endpoint }).then(setNotificationContacts);
```

File: www/front_src/src/Resources/Details/tabs/Notifications/Contacts.tsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+const Contacts = ({
+  contacts,
+  templateColumns,
+  getColumns,
+  headers,
+}: Props): JSX.Element => {
+  const goToUri = (uri: string): void => {
+    window.location.href = uri;
```
[the remark this answers is not in the document]
Comment (Jordan91) [reply www/front_src/src/Resources/Details/tabs/Notifications/Contacts.tsx:26]: Yes as we talk :)  , but actualy if i use the hook useNavigate , the backend must be return or adapt the path for this "uri"  without centreon =>

Now that is return:   
notification_policy: "/centreon/api/latest/configuration/hosts/14/services/19/notification-policy"

[the remark this answers is not in the document]
Comment (Thebarda) [reply www/front_src/src/Resources/Details/tabs/Notifications/Contacts.tsx:26]: This uri does not lead to a contact's page

[the remark this answers is not in the document]
Comment (Jordan91) [reply www/front_src/src/Resources/Details/tabs/Notifications/Contacts.tsx]: actually, I guess I’m a little tired...

[the remark this answers is not in the document]
Comment (Jordan91) [reply www/front_src/src/Resources/Details/tabs/Notifications/Contacts.tsx]: that is the right value return by the back end :
"configuration_uri": "/centreon/main.php?p=60301&o=c&contact_id=14"

[the remark this answers is not in the document]
Comment (Thebarda) [reply www/front_src/src/Resources/Details/tabs/Notifications/Contacts.tsx]: Yeah. Backend folks must adapt the path. But there is a way to handle it on our side

File: lang/es_ES.UTF-8/LC_MESSAGES/messages.po [hunk 5]
```diff
@@ -15017,6 +15017,7 @@ msgstr ""
 # msgid "Login with"
 # msgstr ""
 
+# msgid "You are not entitled to access configuration"
```
Comment (Thebarda) [inline lang/es_ES.UTF-8/LC_MESSAGES/messages.po:15020]: msgstr is missing

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 6]
```diff
@@ -16710,6 +16710,8 @@ msgstr "Se connecter avec"
 msgid "Can't resolve username from login claim using configured regular expression"
 msgstr "Impossible de résoudre le nom d'utilisateur à partir de la déclaration de connexion en utilisant l'expression régulière configurée"
 
+# msgid "You are not entitled to access configuration"
+# msgstr "Vous n’êtes pas autorisé à accéder à la configuration"
```
Comment (Thebarda) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:16714]:
```suggestion
# msgid "You are not entitled to access configuration"
# msgstr "Vous n’êtes pas autorisé à accéder à la configuration"

```

File: lang/pt_BR.UTF-8/LC_MESSAGES/messages.po [hunk 7]
```diff
@@ -15468,6 +15468,8 @@ msgstr ""
 # msgid "Login with"
 # msgstr ""
 
+# msgid "You are not entitled to access configuration"
+# msgstr ""
```
Comment (Thebarda) [inline lang/pt_BR.UTF-8/LC_MESSAGES/messages.po:15472]:
```suggestion
# msgid "You are not entitled to access configuration"
# msgstr ""

```

File: lang/pt_PT.UTF-8/LC_MESSAGES/messages.po [hunk 8]
```diff
@@ -15456,6 +15456,8 @@ msgstr ""
 # msgid "Login with"
 # msgstr ""
 
+# msgid "You are not entitled to access configuration"
+# msgstr ""
```
Comment (Thebarda) [inline lang/pt_PT.UTF-8/LC_MESSAGES/messages.po:15460]:
```suggestion
# msgid "You are not entitled to access configuration"
# msgstr ""

```

File: www/front_src/src/Resources/Details/index.test.tsx [hunk 9]
```diff
@@ -1695,4 +1726,44 @@ describe(Details, () => {
       );
     });
   });
+
+  it.only('displays contacts and contact groups in notification tab when she is clicked', async () => {
```
Comment (Thebarda) [inline www/front_src/src/Resources/Details/index.test.tsx:1730]:
```suggestion
  it('displays contacts and contact groups when the Notification tab is clicked', async () => {
```

File: www/front_src/src/Resources/Details/tabs/Notifications/Contacts.tsx [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+    const iconColor = canGoToConfiguration ? 'primary' : 'default';
+
+    return (
+      <Tooltip title={tooltipTitle}>
+        <IconButton
+          color={iconColor}
+          size="small"
+          sx={{ justifySelf: 'flex-end', marginRight: 1 }}
+          title={t(tooltipTitle)}
+          onClick={(): void => goToUri(configuration_uri)}
```
Comment (Thebarda) [inline www/front_src/src/Resources/Details/tabs/Notifications/Contacts.tsx:47]: Extract it to a specific function

File: www/front_src/src/Resources/Details/tabs/Notifications/Contacts.tsx [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+  contacts,
+  templateColumns,
+  getColumns,
+  headers,
+}: Props): JSX.Element => {
+  const goToUri = (uri): void => {
+    window.location.href = uri as string;
+  };
+
+  const getConfigurationColumn = ({ configuration_uri }): JSX.Element => {
```
Comment (Thebarda) [inline www/front_src/src/Resources/Details/tabs/Notifications/Contacts.tsx:33]: It's recommended to extract into a specific component or memoize it

File: www/front_src/src/Resources/Details/tabs/Notifications/Notification.tsx [hunk 12]
```diff
@@ -0,0 +1,144 @@
...
+  );
+
+  const contactWithEmailHeaders = (
+    <>
+      {contactHeaders}
+      <Typography sx={{ fontWeight: 'bold' }}>{t(labelEmail)}</Typography>
+    </>
+  );
+
+  const getContactColumns = ({ name, alias }): JSX.Element => {
```
Comment (Thebarda) [inline www/front_src/src/Resources/Details/tabs/Notifications/Notification.tsx:70]: Extract it or memoize it

File: www/front_src/src/Resources/Details/tabs/Notifications/Notification.tsx [hunk 13]
```diff
@@ -0,0 +1,144 @@
...
+    return (
+      <>
+        <ContactCell paddingLeft={1}>{name}</ContactCell>
+
+        <ContactCell>{alias}</ContactCell>
+      </>
+    );
+  };
+
+  const getContactWithEmailColumns = (contact: Contact): JSX.Element => {
```
Comment (Thebarda) [inline www/front_src/src/Resources/Details/tabs/Notifications/Notification.tsx:80]: Extract it or memoize it

File: www/front_src/src/Resources/Details/tabs/Notifications/Notification.tsx (hunk 12 above)
Comment (Jordan91) [reply www/front_src/src/Resources/Details/tabs/Notifications/Notification.tsx:70]: On a component i understand what i do , declare component and set only props i would memoize, but for a function , i have some difficulties :/

File: www/front_src/src/Resources/Details/tabs/Notifications/Notification.tsx (hunk 13 above)
Comment (Jordan91) [reply www/front_src/src/Resources/Details/tabs/Notifications/Notification.tsx:80]: On a component i understand what i do , declare component and set only props i would memoize, but for a function , i have some difficulties :/

File: www/front_src/src/Resources/Details/tabs/Notifications/Contacts.tsx (hunk 10 above)
Comment (Jordan91) [reply www/front_src/src/Resources/Details/tabs/Notifications/Contacts.tsx:47]: you talk about this function  goToUri  or onClick ? 

const goToUri = (uri): void => {
    window.location.href = uri as string;
  };

**Review by Thebarda — DISMISSED** [review]

File: www/front_src/src/Resources/Details/tabs/Notifications/Notification.tsx [hunk 14]
```diff
@@ -0,0 +1,147 @@
...
+  labelName,
+  labelNotificationStatus,
+} from '../../../translatedLabels';
+
+import { Contact, ContactGroup, NotificationContacts } from './models';
+import Contacts from './Contacts';
+import ContactsLoadingSkeleton from './ContactsLoadingSkeleton';
+import ContactCell from './ContactCell';
+
+const Notification = (): JSX.Element => {
```
Comment (bdauria) [inline www/front_src/src/Resources/Details/tabs/Notifications/Notification.tsx:30]:
```suggestion
const Notifications = (): JSX.Element => {
```
```suggestion
const Notification = (): JSX.Element => {
```

Comment (bdauria) [inline www/front_src/src/Resources/Details/tabs/Notifications/Notification.tsx:30]:
```suggestion
const Notifications = (): JSX.Element => {
```
```suggestion
const Notification = (): JSX.Element => {
```

File: lang/es_ES.UTF-8/LC_MESSAGES/messages.po [hunk 15]
```diff
@@ -15017,6 +15017,9 @@ msgstr ""
...
```
[the remark this answers is not in the document]
Comment (Jordan91) [reply lang/es_ES.UTF-8/LC_MESSAGES/messages.po]: Yes i made the changes

File: lang/es_ES.UTF-8/LC_MESSAGES/messages.po [hunk 16]
```diff
@@ -15017,6 +15017,9 @@ msgstr ""
 # msgid "Login with"
 # msgstr ""
 
+# msgid "You are not authorized to access configuration"
```
Comment (Thebarda) [inline lang/es_ES.UTF-8/LC_MESSAGES/messages.po:15020]:
```suggestion
# msgid "You are not authorized to access to the configuration"
```

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 17]
```diff
@@ -16710,6 +16710,9 @@ msgstr "Se connecter avec"
 msgid "Can't resolve username from login claim using configured regular expression"
 msgstr "Impossible de résoudre le nom d'utilisateur à partir de la déclaration de connexion en utilisant l'expression régulière configurée"
 
+# msgid "You are not authorized to access configuration"
```
Comment (Thebarda) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:16713]:
```suggestion
# msgid "You are not authorized to access to the configuration"
```

File: lang/pt_BR.UTF-8/LC_MESSAGES/messages.po [hunk 18]
```diff
@@ -15468,6 +15468,9 @@ msgstr ""
 # msgid "Login with"
 # msgstr ""
 
+# msgid "You are not authorized to access configuration"
```
Comment (Thebarda) [inline lang/pt_BR.UTF-8/LC_MESSAGES/messages.po:15471]:
```suggestion
# msgid "You are not authorized to access to the configuration"
```

File: lang/pt_PT.UTF-8/LC_MESSAGES/messages.po [hunk 19]
```diff
@@ -15456,6 +15456,9 @@ msgstr ""
 # msgid "Login with"
 # msgstr ""
 
+# msgid "You are not authorized to access configuration"
```
Comment (Thebarda) [inline lang/pt_PT.UTF-8/LC_MESSAGES/messages.po:15459]:
```suggestion
# msgid "You are not authorized to access to the configuration"
```

**Review by Thebarda — DISMISSED** [review]

File: lang/es_ES.UTF-8/LC_MESSAGES/messages.po [hunk 20]
```diff
@@ -15062,6 +15065,12 @@ msgstr ""
 # msgid "Discard"
 # msgstr ""
 
+# msgid "No contact is configured for this resource"
```
Comment (bdauria) [inline lang/es_ES.UTF-8/LC_MESSAGES/messages.po:15068]:
```suggestion
# msgid "No contacts are configured for this resource"
```
```suggestion
# msgid "No contact is configured for this resource"
```

File: lang/es_ES.UTF-8/LC_MESSAGES/messages.po [hunk 21]
```diff
@@ -15062,6 +15065,12 @@ msgstr ""
...
+# msgstr ""
+
+# msgid "No contact groups is configured for this resource"
```
Comment (bdauria) [inline lang/es_ES.UTF-8/LC_MESSAGES/messages.po:15071]:
```suggestion
# msgid "No contact groups are configured for this resource"
```
```suggestion
# msgid "No contact groups is configured for this resource"
```

File: www/front_src/src/Resources/Details/tabs/Notifications/Notifications.tsx [hunk 22]
```diff
@@ -0,0 +1,184 @@
...
+      );
+    },
+    [],
+  );
+
+  React.useEffect(() => {
+    if (isNil(details)) {
+      return;
+    }
+    loadNotificationContacts();
```
Comment (bdauria) [inline www/front_src/src/Resources/Details/tabs/Notifications/Notifications.tsx:75]:
```suggestion

    loadNotificationContacts();
```

Comment (bdauria) [inline www/front_src/src/Resources/Details/tabs/Notifications/Notifications.tsx:75]:
```suggestion

    loadNotificationContacts();
```

File: www/front_src/src/Resources/Details/tabs/Notifications/Notifications.tsx [hunk 23]
```diff
@@ -0,0 +1,184 @@
...
+      {isEmpty(notificationContacts.contacts) ? (
+        <Paper>
+          <Stack alignItems="center" direction="row" padding={1} spacing={0.5}>
+            <PersonIcon color="primary" fontSize="large" />
+            <Typography sx={{ fontWeight: 'bold' }}>
+              {t(labelContacts)}
+            </Typography>
+          </Stack>
+          <Stack padding={1}>
+            <Typography sx={{ fontWeight: 'bold' }}>
```
Comment (bdauria) [inline www/front_src/src/Resources/Details/tabs/Notifications/Notifications.tsx:126]:
```suggestion
            <Typography>
```

Comment (bdauria) [inline www/front_src/src/Resources/Details/tabs/Notifications/Notifications.tsx:126]:
```suggestion
            <Typography>
```

File: www/front_src/src/Resources/Details/tabs/Notifications/Notifications.tsx [hunk 24]
```diff
@@ -0,0 +1,184 @@
...
+      </Paper>
+      {isEmpty(notificationContacts.contacts) ? (
+        <Paper>
+          <Stack alignItems="center" direction="row" padding={1} spacing={0.5}>
+            <PersonIcon color="primary" fontSize="large" />
+            <Typography sx={{ fontWeight: 'bold' }}>
+              {t(labelContacts)}
+            </Typography>
+          </Stack>
+          <Stack padding={1}>
```
Comment (bdauria) [inline www/front_src/src/Resources/Details/tabs/Notifications/Notifications.tsx:125]: Is a stack surrouning only one element really necessary?

besides that, I think we should center the text indicating that no contacts are available

Comment (bdauria) [inline www/front_src/src/Resources/Details/tabs/Notifications/Notifications.tsx:125]: Is a stack surrouning only one element really necessary?

besides that, I think we should center the text indicating that no contacts are available

File: www/front_src/src/Resources/Details/tabs/Notifications/Notifications.tsx [hunk 25]
```diff
@@ -0,0 +1,184 @@
...
+
+          <Contacts
+            contacts={notificationContacts?.contacts as Array<Contact>}
+            getColumns={getContactWithEmailColumns}
+            headers={contactWithEmailHeaders}
+            templateColumns="1fr 1fr 1fr 1fr"
+          />
+        </Stack>
+      )}
+      {isEmpty(notificationContacts.contacts) ? (
```
Comment (bdauria) [inline www/front_src/src/Resources/Details/tabs/Notifications/Notifications.tsx:148]: This should be reused as a dedicated Component to avoid duplications

Comment (bdauria) [inline www/front_src/src/Resources/Details/tabs/Notifications/Notifications.tsx:148]: This should be reused as a dedicated Component to avoid duplications

**Review by Thebarda — DISMISSED** [review]

**Review by Thebarda — DISMISSED** [review]

**Review by jdelpierre — DISMISSED** [review]
OK FE part

File: lang/es_ES.UTF-8/LC_MESSAGES/messages.po [hunk 26]
```diff
@@ -15017,6 +15017,9 @@ msgstr ""
 # msgid "Login with"
 # msgstr ""
 
+# msgid "You are not authorized to access to the configuration"
```
Comment (bdauria) [inline lang/es_ES.UTF-8/LC_MESSAGES/messages.po:15020]: You are not authorized to access the configuration

File: www/front_src/src/Resources/Details/tabs/Notifications/Contacts.tsx [hunk 27]
```diff
@@ -0,0 +1,119 @@
...
+} from '../../../translatedLabels';
+import memoizeComponent from '../../../memoizedComponent';
+
+import { ContactGroup, Contact } from './models';
+
+interface Props {
+  contacts: Array<Contact> | Array<ContactGroup> | undefined;
+  getColumns: (contact) => JSX.Element;
+  headers: JSX.Element;
+  messageNoContacts: JSX.Element;
```
Comment (bdauria) [inline www/front_src/src/Resources/Details/tabs/Notifications/Contacts.tsx:28]: noContactsMessage

**Review by Thebarda — DISMISSED** [review]

**Review by jdelpierre — DISMISSED** [review]
OK FE Part

File: doc/API/centreon-api-v22.04.yaml [hunk 28]
```diff
@@ -674,6 +674,70 @@ paths:
...
+            application/json:
+              schema:
+                $ref: './NotificationPolicy.yaml'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        '500':
+          $ref: '#/components/responses/InternalServerError'
+  /configuration/{host_id}/services/{service_id}/notification-policy:
```
Comment (a-launois) [inline doc/API/centreon-api-v22.04.yaml:698]:
```suggestion
  /configuration/hosts/{host_id}/services/{service_id}/notification-policy:
```

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 29]
```diff
@@ -16767,6 +16770,12 @@ msgstr "Il y a des erreurs dans le formulaire. Voulez-vous quitter le formulaire
 msgid "Discard"
 msgstr "Annuler"
 
+msgid "No contacts are configured for this resource"
+msgstr "Aucun contacts configurés pour cette ressource"
+
+msgid "No contact groups are configured for this resource"
+msgstr "Aucun contact groupes configurés pour cette ressource"
+
```
Comment (a-launois) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:16778]:
```suggestion
msgid "No contacts are configured for this resource"
msgstr "Aucun contact configuré pour cette ressource"

msgid "No contact groups are configured for this resource"
msgstr "Aucun contact groupe configuré pour cette ressource"

```

**Review by a-launois — APPROVED** [review]

**Review by jdelpierre — APPROVED** [review]

**Review by Thebarda — APPROVED** [review]

**Pull request merged**